src/core/src/package_managers/YumPackageManager.py

```diff
@@ -347,26 +386,249 @@ def get_package_size(self, output):
+                #     if self.enable_on_reboot_value or self.apply_updates_value == 'yes':
+                #         break
+
+        if self.apply_updates_value == 'no' and not self.enable_on_reboot_value:
+            return Constants.AutomaticOSPatchStates.DISABLED
+        elif self.apply_updates_value == 'yes' or self.enable_on_reboot_value:
+            return Constants.AutomaticOSPatchStates.ENABLED
+        else:
+            return Constants.AutomaticOSPatchStates.UNKNOWN
+
+    def __init_auto_update_for_yum_cron(self):
+        """ Initializes all generic auto OS update variables with the config values for yum cron service """
+        self.os_patch_configuration_settings_file_path = self.yum_cron_configuration_settings_file_path
+        self.download_updates_identifier_text = self.yum_cron_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.yum_cron_apply_updates_identifier_text
+        self.auto_update_config_pattern_match_text = self.yum_cron_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.yum_cron_enable_on_reboot_check_cmd
+
+    def __init_auto_update_for_dnf_automatic(self):
+        """ Initializes all generic auto OS update variables with the config values for dnf automatic service """
+        self.os_patch_configuration_settings_file_path = self.dnf_automatic_configuration_file_path
+        self.download_updates_identifier_text = self.dnf_automatic_download_updates_identifier_text
+        self.apply_updates_identifier_text = self.dnf_automatic_apply_updates_identifier_text
+        self.auto_update_config_pattern_match_text = self.dnf_automatic_config_pattern_match_text
+        self.enable_on_reboot_check_cmd = self.dnf_automatic_enable_on_reboot_check_cmd
 
     def disable_auto_os_update(self):
-        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
-        # NOTE: Implementation pending
-        pass
+        """ Disables auto OS updates on the machine only if they are enable_on_reboot and logs the default settings the machine comes with """
+        try:
+            self.composite_logger.log_debug("Disabling auto OS updates in all identified services...")
+            for os_auto_update_service in self.auto_os_update_service_list:
+                # todo: Better way to implement this? Keeping functions separate since not all services have the same steps to disable and identifying the service specific commands is easier this way
+                if os_auto_update_service == self.yum_cron:
+                    self.disable_auto_os_update_for_yum_cron()
+                elif os_auto_update_service == self.dnf_automatic:
+                    self.disable_auto_os_update_for_dnf_automatic()
+                # elif os_auto_update_service == self.packagekit:
+                #     self.disable_auto_os_update_for_packagekit()
+                else:
```

Comment:
There's no identification happening here - it's just processing a static list. Each function is called exactly once and the error is thrown once (in this implementation). It's cleaner to just call each function once.

Reply:
Resolved